File: VB/Applications/WinFormsLog4NetVBDI/Forms/MainForm.vb
Imports LogViewerVB.WinForms

Public Class MainForm

#Region "Constructors"

    Sub New(controlsDataStore As MainControlsDataStore)

        ' This call is required by the designer.
        InitializeComponent()

        ' wire up the control
        HostPanel.AddControl(controlsDataStore.LogViewer)

    End Sub

#End Region

End Class

File: CSharp/Applications/WinFormsNLogDI/Forms/MainForm.cs
using LogViewer.WinForms;
using WinFormsNLogDI.DataStores;

namespace WinFormsNLogDI;

public partial class MainForm : Form
{
    #region Constructors

    public MainForm(MainControlsDataStore controlsDataStore)
    {
        InitializeComponent();

        // wire up the control
        HostPanel.AddControl(controlsDataStore.LogViewer);
    }

    #endregion
}